internal/handlers/pg/msg_getlog.go

```diff
@@ -36,55 +39,114 @@ func (h *Handler) MsgGetLog(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		return nil, lazyerrors.Error(err)
 	}
 
-	pv, _, _ = strings.Cut(pv, " ")
-	mv := version.Get()
-
-	var log types.Array
-	for _, line := range []string{
-		"Powered by 🥭 FerretDB " + mv.Version + " and PostgreSQL " + pv + ".",
-		"Please star us on GitHub: https://github.com/FerretDB/FerretDB",
-	} {
-		b, err := json.Marshal(map[string]any{
-			"msg":  line,
-			"tags": []string{"startupWarnings"},
-			"s":    "I",
-			"c":    "STORAGE",
-			"id":   42000,
-			"ctx":  "initandlisten",
-			"t": map[string]string{
-				"$date": time.Now().UTC().Format("2006-01-02T15:04:05.999Z07:00"),
-			},
-		})
+	if _, ok := getLog.(string); !ok {
+		return nil, common.NewErrorMsg(common.ErrTypeMismatch, "Argument to getLog must be of type String")
+	}
+
+	var resDoc *types.Document
+	switch getLog {
+	case "*":
+		resDoc = must.NotFail(types.NewDocument(
+			"names", must.NotFail(types.NewArray("global", "startupWarnings")),
+			"ok", float64(1),
+		))
+
+	case "global":
+		log, err := requirRecordsLog(zapcore.DebugLevel)
 		if err != nil {
 			return nil, lazyerrors.Error(err)
 		}
-		if err = log.Append(string(b)); err != nil {
+		resDoc = must.NotFail(types.NewDocument(
+			"log", &log,
+			"totalLinesWritten", int64(log.Len()),
+			"ok", float64(1),
+		))
+
+	case "startupWarnings":
+		var pv string
+		err = h.pgPool.QueryRow(ctx, "SHOW server_version").Scan(&pv)
+		if err != nil {
 			return nil, lazyerrors.Error(err)
 		}
-	}
+		pv, _, _ = strings.Cut(pv, " ")
+		mv := version.Get()
 
-	var reply wire.OpMsg
-	err = reply.SetSections(wire.OpMsgSection{
-		Documents: []*types.Document{must.NotFail(types.NewDocument(
-			"totalLinesWritten", int32(log.Len()),
+		var log types.Array
+		for _, line := range []string{
+			"Powered by 🥭 FerretDB " + mv.Version + " and PostgreSQL " + pv + ".",
+			"Please star us on GitHub: https://github.com/FerretDB/FerretDB",
+		} {
+			b, err := json.Marshal(map[string]any{
+				"msg":  line,
+				"tags": []string{"startupWarnings"},
+				"s":    "I",
+				"c":    "STORAGE",
+				"id":   42000,
+				"ctx":  "initandlisten",
+				"t": map[string]string{
+					"$date": time.Now().UTC().Format("2006-01-02T15:04:05.999Z07:00"),
+				},
+			})
+			if err != nil {
+				return nil, lazyerrors.Error(err)
+			}
+			if err = log.Append(string(b)); err != nil {
+				return nil, lazyerrors.Error(err)
+			}
+		}
+		resDoc = must.NotFail(types.NewDocument(
 			"log", &log,
+			"totalLinesWritten", int64(log.Len()),
 			"ok", float64(1),
-		))},
-	})
+		))
+
+	default:
+		errMsg := fmt.Sprintf("no RamLog named: %s", getLog)
+		return nil, common.NewErrorMsg(0, errMsg)
+	}
+
+	var reply wire.OpMsg
+	err = reply.SetSections(wire.OpMsgSection{Documents: []*types.Document{resDoc}})
 	if err != nil {
 		return nil, lazyerrors.Error(err)
 	}
 
 	return &reply, nil
 }
+
+// requirRecordsLog returns an array of records from logging buffer with given level.
+func requirRecordsLog(level zapcore.Level) (log types.Array, err error) {
```

Comment:
```suggestion
func requirRecordsLog(level zapcore.Level) (*types.Array, error) {
```

We should always accept and return composite types by pointers. We also don't use named result values unless necessary